src/libsync/propagatedownload.cpp

```diff
@@ -344,6 +344,42 @@ void PropagateDownloadFile::start()
         }
     }
 
+    // If we have a conflict where size and mtime are identical,
+    // compare the remote checksum to the local one.
+    // Maybe it's not a real conflict and no download is necessary!
+    QByteArray checksumType = parseChecksumHeaderType(_item->_checksumHeader);
+    if (_item->_instruction == CSYNC_INSTRUCTION_CONFLICT
+        && _item->_size == _item->log._other_size
+        && _item->_modtime == _item->log._other_modtime
+        && collisionSafeHashes().contains(checksumType)) {
```

Comment:
So you accept collision without a hash (from the reconsile phase),   but not if adler32 is used?